express/index.cjs

```diff
@@ -383,22 +383,22 @@ app.get('/api/jobs', async (req, res) => {
 // forget password, should send email to link
 app.post('/forgot-password', async (req, res) => {
   const newTime = new Date(Date.now());
+  writer.write(`${setTimestamp(newTime)} || source: /forget-password\n`);
   const { email } = req.body;
 
   try {
     if (!email) {
-      return res.status(400).json({ success: false, error: 'Missing email' });
+      throw({status: 400, success: false, error: 'failed reset attempt: password', reason: 'missing field'});
     }
 
      // Use checkUser helper function
      const user = await checkUser(req, email);
 
      if (!user.exists) {
-         return res.status(404).json({ success: false, error: 'Email not found' });
+      throw({status: 404, success: false, error: 'failed reset attempt: password', reason: 'email not found'});
      }
 
     const { usertype, userId } = user;
-    // const userIdHex = userId.toString('hex'); // Convert binary ID to hexadecimal string
 
     // Create a reset token
     const resetToken = jwt.sign({email, id: userId, type: usertype}, process.env.JWT_KEY, { expiresIn: '1h' });
```

Comment:
The email isn't assigned properly, and may not be used in the JWT. Please check this.

Reply:
would it supposed to be assigned like this: `const resetToken = jwt.sign({email: email, id: userId, type: usertype}` ??
I decoded the jwt and the email is decoded along with the id and type